handlers.go

```diff
@@ -0,0 +1,300 @@
+package main
+
+import (
+	"bytes"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"regexp"
+	"time"
+)
+
+const maxFilterTasks = 20
+
+func handleTask(w http.ResponseWriter, req *http.Request) {
+
+	switch req.Method {
+	case http.MethodPost:
+		writeTask(w, req, true)
+	case http.MethodGet:
+		getTask(w, req)
+	case http.MethodPut:
+		writeTask(w, req, false)
+	case http.MethodDelete:
+		deleteTask(w, req)
+	}
+}
+
+func handleTaskDone(w http.ResponseWriter, req *http.Request) {
+	id := req.URL.Query().Get("id")
+	if id == "" {
+		setError(w, "Не указан идентификатор", http.StatusBadRequest)
+		return
+	}
+
+	task, err := getTaskByID(id)
+	if err != nil {
+		status := http.StatusBadRequest
+		if err == sql.ErrNoRows {
```

Comment:
Ошибки лучше так сравнивать
```suggestion
		if errors.Is(err, sql.ErrNoRows) {
```

Reply:
Ок, учту на будущее.